src/js/contexts/AnnounceContext/AnnounceContext.js

```diff
@@ -1,33 +1,84 @@
 import React from 'react';
 import { AnnounceContextPropTypes } from './propTypes';
 
+/**
+ * Creates a hidden announcer element for screen reader accessibility
+ * This element is used to announce dynamic content changes to screen readers
+ */
 const createAnnouncer = () => {
   const announcer = document.createElement('div');
-  announcer.id = 'grommet-announcer';
-  announcer.style.left = '-100%';
-  announcer.style.right = '100%';
-  announcer.style.position = 'fixed';
-  announcer.style['z-index'] = '-1';
+  announcer.setAttribute('id', 'grommet-announcer');
+  announcer.setAttribute('aria-live', 'polite');
+  announcer.setAttribute('aria-atomic', 'true');
 
+  // Position element off-screen to hide it visually
+  announcer.style.position = 'absolute';
+  announcer.style.left = '-9999px';
+  announcer.style.height = '1px';
+  announcer.style.width = '1px';
+  announcer.style.overflow = 'hidden';
+
+  // Add to DOM at the beginning of body for best screen reader support
+  document.body.appendChild(announcer);
   document.body.insertBefore(announcer, document.body.firstChild);
 
   return announcer;
 };
 
 export const AnnounceContext = React.createContext(
   (message, mode = 'polite', timeout = 500) => {
-    // we only create a new container if we don't have one already
-    // we create a separate node so that grommet does not set aria-hidden to it
+    // Get existing announcer or create a new one
     const announcer =
-      document.body.querySelector(`#grommet-announcer[aria-live]`) ||
-      createAnnouncer();
+      document.body.querySelector('#grommet-announcer') || createAnnouncer();
+
+    // Clear any existing timeouts to prevent overlapping announcements
+    if (announcer.dataset.initialAnnounceTimeoutId) {
+      clearTimeout(Number(announcer.dataset.initialAnnounceTimeoutId));
+      delete announcer.dataset.initialAnnounceTimeoutId;
+    }
+    if (announcer.dataset.timeoutId) {
+      clearTimeout(Number(announcer.dataset.timeoutId));
+      delete announcer.dataset.timeoutId;
+    }
 
-    announcer.setAttribute('aria-live', 'off');
-    announcer.innerHTML = message;
     announcer.setAttribute('aria-live', mode);
-    setTimeout(() => {
-      announcer.innerHTML = '';
-    }, timeout);
+
+    // have to reset the content with something different (non breaking space)
+    // to force the screen reader to recognize the change.
+    announcer.textContent = '\u00A0';
+
+    // Force DOM reflow to ensure screen readers detect the change
```

Comment:
Using JavaScript, you can dynamically change the content inside the element with role="alert". Note that if you need to fire the same alert multiple times (i.e., the content you're dynamically inserting is the same as before), this generally won't be seen as a change and will not lead to an announcement. For this reason, it's usually best to briefly "clear" the contents of the alert container before then injecting the alert message.



from MDN